src/cluster_linearize.h

```diff
@@ -662,6 +678,697 @@ class AncestorCandidateFinder
+ *     candidates, the chunk with the highest-index transaction involving a relevant dependency is
+ *     picked (this will be changed in a later commit).
+ *
+ * - How to decide what dependency to activate (when merging chunks):
+ *   - After picking two chunks to be merged (see above), the dependency with the lowest-index
+ *     transaction in the other chunk is activated (this will be changed in a later commit).
+ *
+ * - How to decide which chunk to find a dependency to split in:
+ *   - The chunk with the lowest-index representative (an implementation detail) that can be split
+ *     is picked (this will be changed in a later commit).
+ *
+ * - How to decide what dependency to deactivate (when splitting chunks):
+ *   - Inside the selected chunk (see above), among the dependencies whose top feerate is strictly
+ *     higher than its bottom feerate in the selected chunk, if any, the one with the lowest-index
+ *     child is deactivated (this will be changed in a later commit).
+ */
+template<typename SetType>
+class SpanningForestState
+{
+private:
+    /** Data type to represent indexing into m_tx_data. */
+    using TxIdx = uint32_t;
+    /** Data type to represent indexing into m_dep_data. */
+    using DepIdx = uint32_t;
+
+    /** Structure with information about a single transaction. For transactions that are the
+     *  representative for the chunk they are in, this also stores chunk information. */
+    struct TxData {
+        /** The dependencies to parents of this transaction. Immutable after construction. */
+        std::vector<DepIdx> parent_deps;
+        /** The dependencies to children of this transaction. Immutable after construction. */
+        std::vector<DepIdx> child_deps;
+        /** The set of parent transactions of this transaction. Immutable after construction. */
+        SetType parents;
+        /** The set of child transactions of this transaction. Immutable after construction. */
+        SetType children;
+        /** Which transaction holds the chunk_setinfo for the chunk this transaction is in
+         *  (the representative for the chunk). */
+        TxIdx chunk_rep;
+        /** (Only if this transaction is the representative for the chunk it is in) the total
+         *  chunk set and feerate. */
+        SetInfo<SetType> chunk_setinfo;
+    };
+
+    /** Structure with information about a single dependency. */
+    struct DepData {
+        /** Whether this dependency is active. */
+        bool active;
+        /** What the parent and child transactions are. Immutable after construction. */
+        TxIdx parent, child;
+        /** (Only if this dependency is active) the would-be top chunk and its feerate that would
+         *  be formed if this dependency were to be deactivated. */
+        SetInfo<SetType> top_setinfo;
+    };
+
+    /** The set of all transactions in the cluster. */
+    SetType m_transactions;
```

Comment:
c2eaf22503f7af057599f02ecb5f24a267ea33e2

suggestion:
```
-    /** The set of all transactions in the cluster. */
-    SetType m_transactions;
-    /** Information about each transaction (and chunks). Indexed by TxIdx. */
+    /** The set of all TxIdx's of transactions in the cluster indexing into m_tx_data. */
+    SetType m_transaction_idxs;
+    /** Information about each transaction (and chunks). Keeps the "holes" from DepGraph
+        during construction. Indexed by TxIdx. */
     std::vector<TxData> m_tx_data;
```

Reply:
Done, except renaming to `m_transaction_idxs`, because I'm too lazy to update all the affected commits right now. Will leave this open.